universum/modules/vcs/perforce_vcs.py

```diff
@@ -476,7 +476,12 @@ def check_diff_for_depot(self, depot: str) -> str:
 
     def calculate_file_diff(self) -> List[Dict[str, str]]:
         action_list: Dict[str, str] = {}
-        for entry in self.p4.run_opened():
+        try:
+            opened_files = self.p4.run_opened()
+        except P4Exception:
+            return [{'failure': "Getting diff failed due to server error!"}]
```

Comment:
In case of fatal failure, for example, if failure persists even after all retries, return empty list.
Do not return fake data to clients.